tests/functional/flows/test_start_flow.py

```diff
@@ -0,0 +1,122 @@
+    run_monitor_identities = [
+        {
+            "username": "snaggle@jellystone.park",
+            "name": "Snagglepuss",
+            "id": run_monitors[0].split(":")[-1],
+            "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+            "organization": "Hanna-Barbera",
+            "status": "used",
+            "email": "snagglepuss@jellystone.park",
+        },
+        {
+            "username": "yakky@jellystone.park",
+            "name": "Yakky Doodle",
+            "id": run_monitors[1].split(":")[-1],
+            "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+            "organization": "Hanna-Barbera",
+            "status": "used",
+            "email": "yakky@jellystone.park",
+        },
+    ]
+    load_response(
+        RegisteredResponse(
+            service="auth",
+            path="/v2/api/identities",
+            json={
+                "identities": [
+                    owner_identity,
+                    *run_manager_identities,
+                    *run_monitor_identities,
+                ],
+            },
+        )
+    )
+
+    # Construct the command line.
+    arguments = [f"'{flow_id}'", "--input", f"'{json.dumps(body)}'"]
+    for run_manager in run_managers:
+        arguments.extend(("--manager", f"'{run_manager}'"))
+    for run_monitor in run_monitors:
+        arguments.extend(("--monitor", f"'{run_monitor}'"))
+    for tag in tags:
+        arguments.extend(("--tag", f"'{tag}'"))
+    if label is not None:
+        arguments.extend(("--label", f"'{label}'"))
+
+    result = run_line(f"globus flows start {' '.join(arguments)}")
+
+    # all fields present
+    expected_fields = {
+        "Flow ID",
+        "Flow title",
+        "Run ID",
+        "Run label",
+        "Run owner",
+        "Run managers",
+        "Run monitors",
+        "Run tags",
+    }
+    actual_fields = set(re.findall(r"^[\w ]+(?=:)", result.output, flags=re.M))
+    assert expected_fields == actual_fields, "Expected and actual field sets differ"
+
+    tag_match = re.search(r"^Run tags:\s+(?P<tags>.+)$", result.output, flags=re.M)
+    assert tag_match is not None
+    assert ", ".join(tags) in tag_match.group("tags")
+
+    owner_match = re.search(r"^Run owner:\s+(?P<owner>.+)$", result.output, flags=re.M)
+    assert owner_match is not None
+    assert owner_match.group("owner") == owner_identity["username"]
+
+    managers_match = re.search(
+        r"^Run managers:\s+(?P<managers>.+)$", result.output, flags=re.M
+    )
+    assert managers_match is not None
+    assert managers_match.group("managers") == ", ".join(
+        identity["username"] for identity in run_manager_identities
+    )
+
+    monitors_match = re.search(
+        r"^Run monitors:\s+(?P<monitors>.+)$", result.output, flags=re.M
+    )
+    assert monitors_match is not None
+    assert monitors_match.group("monitors") == ", ".join(
+        identity["username"] for identity in run_monitor_identities
+    )
```

Comment:
👍 for the way these matches are done on the output. I like this better than an `assert output == "foo: bar\nbar:baz\n"` style exact-match.

I wrote a helper for this at one point (https://github.com/globus/globus-cli/blob/ae17da9ce9d782a4f8d3a73fe1ed9a948a357264/tests/conftest.py#L156) which I would _not_ recommend  using. It would be good to think of some pattern which we want to encourage, similar to what you have here, and codify it in some better-authored utility class. That way, our tests will consistently check the fields in the text output.

Reply:
Thank you! Yep, it may be beneficial to package output matching in a reusable way.